scripts/pull_translations.rb

```diff
@@ -0,0 +1,167 @@
+# codebase.
+#
+# It will:
+# 1. determine which new keys have been added locally
+# 2. pull translations from PhraseApp
+# 3. generate new .strings files:
+#    - any strings not included the default strings.xml are deleted
+#    - any existing translations that have changed on PhraseApp will be updated
+#    - any new local keys that have translations on PhraseApp will be inserted
+
+def remove_translation_and_comment(node)
+  if node.previous.previous.comment?
+    if node.previous.previous.previous.text?
+      node.previous.previous.previous.unlink
+    end
+    node.previous.previous.unlink
+  end
+  node.previous.unlink if node.previous.text?
+  node.unlink
+end
+
+def remove_excess_strings(filename, included_strings, show_info)
+  puts '▸ Removing unused translated strings'
+  strings_doc = File.open(filename) { |f| Nokogiri::XML(f) }
+  translation_nodes = strings_doc.xpath('//string').to_a
+  all_strings = strings_doc.xpath('//string/@name').to_a.to_set(&:text)
+  excess_strings = all_strings - included_strings
+  translation_nodes.each do |translation|
+    if excess_strings.include? translation['name']
+      remove_translation_and_comment(translation)
+      puts "Removed translation %s" % translation if show_info
+    end
+  end
+  File.write(filename, strings_doc.to_xml)
+end
+
+Struct.new(
+  'Translation',
+  :translated_string,
+  :description_comment,
+  :string_node
+)
+
+def make_translations_hash(filename, included_strings)
+  strings_doc = File.open(filename) { |f| Nokogiri::XML(f) }
+  translation_nodes = strings_doc.xpath('//string').to_a
+  translations_hash = {}
+  translation_nodes.each do |translation|
+    translation_comment = nil
+    if translation.previous.previous.comment?
+      translation_comment = translation.previous.previous
+    end
+    if included_strings.include? translation['name']
+      translations_hash[translation['name']] = Struct::Translation.new(
+        translation.content, translation_comment, translation
+      )
+    end
+  end
+  translations_hash
+end
+
+def replace_updated_translations(filename, translations_hash, show_info)
+  puts '▸ Updating strings that have changed'
+  strings_doc = File.open(filename) { |f| Nokogiri::XML(f) }
+  translation_nodes = strings_doc.xpath('//string').to_a
+  translation_nodes.each do |translation|
+    next unless translations_hash.key?(translation['name'])
+    key = translation['name']
+    next unless !translation.previous.previous.nil? &&
+                translation.previous.previous.comment?
+    translation_comment = translation.previous.previous
+    unless translation_comment.text.eql? translations_hash[key].description_comment.text
+      puts "Updating translation comment for %s" % translation_comment if show_info
+      translation_comment.replace(translations_hash[key].description_comment)
+    end
+    unless translation.content.eql? translations_hash[key].translated_string
```

Comment:
You should probably use `a == b` instead of `a.eql? b`. See [the docs](http://ruby-doc.org/core-2.4.2/Object.html#method-i-eql-3F) for more information about different types of equality in Ruby. I think they're equivalent in this case, but `a == b` is more idiomatic.